dask/array/tests/test_dispatch.py

```diff
@@ -185,8 +191,6 @@ def test_binary_operation_type_precedence(op, arr_upcast, arr_downcast):
         (EncapsulateNDArray(np.arange(4)), True),
         (np.ma.masked_array(np.arange(4), [True, False, True, False]), True),
         (np.arange(4), True),
-        (0.0, True),
-        (0, True),
```

Comment:
Is there a reason why these cases were removed? If we are changing the behavior of `is_valid_array_chunk` to not include scalars, then at least we should test that by making these `False`.

Reply:
Since `float, int` are not valid chunks, but they are still handled by `dask`, it could be counter intuitive to have them here, but if you like to have them here, I can add them back with a comment.

Reply:
Brought them back in 1ce03b5

Reply:
Ah, I see your point. Though, since this is an explicit test of the `is_valid_array_chunk` function (and not a higher-level usage/integration), I'd like to still have them here to confirm the expected behavior of the underlying function, especially since it has changed since the original implementation (which in retrospect didn't correspond well with the function name). Perhaps also adding `("", False)` for a `str` test would be useful as well.

Reply:
@jthielen added.